src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

There is a potential security vulnerability in decoding the JWT token without proper error handling.
```typescript
11:     const token = req.headers?.authorization?.split(' ')[1] || '';
12:     const userDetail = this.jwt.decode(token) || {};
```

#### Suggested Fix
Ensure proper error handling when decoding the JWT token to handle cases where the token is invalid or cannot be decoded.
```typescript
11:     const token = req.headers?.authorization?.split(' ')[1] || '';
12:     let userDetail = {};
13:     try {
14:         userDetail = this.jwt.decode(token) || {};
15:     } catch (error) {
16:         // Handle the error, e.g., log it or set userDetail to a default value
17:     }
```


<!-- This is an auto-generated comment by PullProofAI -->